### PR TITLE
CDI-738 Generate Jakarta EE version of CDI API

### PR DESCRIPTION
**Pull request by antoinesd**
Changing GAV for Jakarta EE.
Usage of Jakarta EE staging repo for EL and Interceptor API (please use -Pstaging when building)
Publication targeted to Jakarta EE repository.

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -19,16 +19,16 @@
...
-		<version>36</version>
+		<groupId>org.eclipse.ee4j</groupId>
+		<artifactId>project</artifactId>
+		<version>1.0.5</version>
 		<relativePath />
 	</parent>
 
-	<groupId>javax.enterprise</groupId>
+	<groupId>jakarta.enterprise</groupId>
 	<artifactId>cdi-api</artifactId>
```
Comment (gunnarmorling) [inline api/pom.xml]: Shouldn't this be `jakarta.cdi-api`, following the pattern of the other specs at https://wiki.eclipse.org/New_Maven_Coordinates?

Comment (manovotn) [reply api/pom.xml]: Oracle's old artifacts had `javax` prefix in their `artifactId` whereas CDI never did.
Though I agree it makes sense to have similar syntax as all other Jakarta artifacts.

Comment (antoinesd) [reply api/pom.xml]: I get your point but it was the coordinate proposed by Dmitry in his first mail and other `enterprise` spec seems to use this as well as you check un the staging repo: https://oss.sonatype.org/content/groups/staging/jakarta/

Comment (manovotn) [reply api/pom.xml]: Not sure I understand you, all the artifacts in the [repo you mentioned](https://oss.sonatype.org/content/groups/staging/jakarta/) are using that prefix :)

Comment (antoinesd) [reply api/pom.xml]: But as other specs in `enterprise` hierarchy follow include `jakarta` in their named, we probably should follow this schema. that would give:

        <groupId>jakarta.enterprise</groupId>
        <artifactId>jakarta.enterprise.cdi-api</artifactId>

Wdyt?

Comment (gunnarmorling) [reply api/pom.xml]: Better double check with Dmitry then what he meant. In fact following the example of the other "enterprise" specs (not that I'd know what that even means), it even should be `jakarta.enterprise.cdi-api` probably.

For BV, we'll use jakarta.validation-api as the artifact id (we also just had validation-api before, but changed it to be consistent with the others).

Comment (gunnarmorling) [reply api/pom.xml]: Ha, cross-posted, but we arrived at the same conclusion 👍

Comment (manovotn) [reply api/pom.xml]: Yea, that sounds just about right

Comment (antoinesd) [reply api/pom.xml]: Done!

**Pull request merged**